android/modules/ui/src/java/ti/modules/titanium/ui/widget/searchbar/TiUISearchBar.java

```diff
@@ -1,180 +1,354 @@
+			searchView.setIconifiedByDefault(TiConvert.toBoolean(properties, TiC.PROPERTY_ICONIFIED_BY_DEFAULT, false));
+		}
+		updateCloseButton();
+		updateInputType();
+
+		// Let base class handle all other properties.
+		super.processProperties(properties);
 	}
 
 	@Override
 	public void propertyChanged(String key, Object oldValue, Object newValue, KrollProxy proxy)
 	{
+		// Validate.
+		if (key == null) {
+			return;
+		}
+
+		// Fetch the search view.
+		SearchView searchView = getSearchView();
+		if (searchView == null) {
+			return;
+		}
+
+		// Handle property change.
 		if (key.equals(TiC.PROPERTY_SHOW_CANCEL)) {
-			boolean showCancel = TiConvert.toBoolean(newValue);
-			cancelBtn.setVisibility(showCancel ? View.VISIBLE : View.GONE);
+			updateCloseButton();
 		} else if (key.equals(TiC.PROPERTY_BAR_COLOR)) {
-			nativeView.setBackgroundColor(TiConvert.toColor(TiConvert.toString(newValue)));
-		} else if (key.equals(TiC.PROPERTY_PROMPT)) {
-			String strPrompt = TiConvert.toString(newValue);
-			promptText.setText(strPrompt);
-		} else if (key.equals(TiC.PROPERTY_BACKGROUND_IMAGE)) {
-			processBackgroundImage(newValue, proxy);
+			searchView.setBackgroundColor(TiConvert.toColor(TiConvert.toString(newValue)));
+		} else if (key.equals(TiC.PROPERTY_COLOR)) {
+			EditText editText = getEditText();
+			if (editText != null) {
+				editText.setTextColor(TiConvert.toColor(TiConvert.toString(newValue)));
+			}
+		} else if (key.equals(TiC.PROPERTY_VALUE)) {
+			boolean wasIgnored = this.isIgnoringChangeEvent;
+			this.isIgnoringChangeEvent = true;
+			searchView.setQuery(TiConvert.toString(newValue), false);
+			this.isIgnoringChangeEvent = wasIgnored;
+		} else if (key.equals(TiC.PROPERTY_HINT_TEXT)) {
+			searchView.setQueryHint(TiConvert.toString(newValue));
+		} else if (key.equals(TiC.PROPERTY_HINT_TEXT_COLOR)) {
+			EditText editText = getEditText();
+			if (editText != null) {
+				editText.setHintTextColor(TiConvert.toColor(TiConvert.toString(newValue)));
+			}
+		} else if (key.equals(TiC.PROPERTY_ICONIFIED)) {
+			searchView.setIconified(TiConvert.toBoolean(newValue, false));
+		} else if (key.equals(TiC.PROPERTY_ICONIFIED_BY_DEFAULT)) {
+			searchView.setIconifiedByDefault(TiConvert.toBoolean(newValue, false));
+		} else if (key.equals(TiC.PROPERTY_AUTOCAPITALIZATION) || key.equals(TiC.PROPERTY_AUTOCORRECT)) {
+			updateInputType();
 		} else {
 			super.propertyChanged(key, oldValue, newValue, proxy);
 		}
 	}
 
-	private void processBackgroundImage(Object imgValue, KrollProxy proxy)
+	@Override
+	public void focus()
+	{
+		// Fetch the search view.
+		SearchView searchView = getSearchView();
+		if (searchView == null) {
+			return;
+		}
+
+		// Expand search view if currently iconified.
+		if (searchView.isIconified()) {
+			searchView.setIconified(false);
+		}
+
+		// Give this view the focus.
+		super.focus();
+
+		// Show the virtual keyboard.
+		InputMethodManager inputManager =
```

Comment:
Since this is used a couple of times, could grab input manager in the constructor and re-use it